explorer/src/utils/coingecko.tsx

```diff
@@ -87,3 +89,137 @@ export function useCoinGecko(coinId?: string): CoinGeckoResult | undefined {
 
   return coinInfo;
 }
+
+export type CoinGeckoToken = {
+  id: string;
+  symbol: string;
+  name: string;
+  image: string;
+  current_price: number;
+  market_cap: number;
+  market_cap_rank: number;
+  fully_diluted_valuation: number;
+  total_volume: number;
+  high_24h: number;
+  low_24h: number;
+  price_change_24h: number;
+  price_change_percentage_24h: number;
+  market_cap_change_24h: number;
+  market_cap_change_percentage_24h: number;
+  circulating_supply: number;
+  total_supply: number;
+  max_supply: number;
+  ath: number;
+  ath_change_percentage: number;
+  ath_date: Date;
+  atl: number;
+  atl_change_percentage: number;
+  atl_date: Date;
+  last_updated: Date;
+  sparkline_in_7d: {
+    price: Array<number>;
+  };
+  price_change_percentage_1h_in_currency: number;
+  price_change_percentage_24h_in_currency: number;
+  price_change_percentage_7d_in_currency: number;
+};
+
+export function useCoinGeckoCategoryTokens(
+  categoryId: string
+): [Error | null, Boolean, Array<CoinGeckoToken>] {
+  const [tokens, setTokens] = React.useState<Array<CoinGeckoToken>>([]);
+  const [loading, setLoading] = React.useState(false);
+  const [error, setError] = React.useState(null);
+
+  React.useEffect(() => {
+    let interval: NodeJS.Timeout | undefined;
+    const fetchSolanaStats = () => {
+      setError(null);
+      setLoading(true);
```

Comment:
for calls after initial load don't set loading to avoid flashing on the page.

Reply:
Done